spec/unit/shoulda/matchers/active_model/allow_value_matcher_spec.rb

```diff
@@ -315,4 +315,49 @@
     end
   end
 
+  context 'when the attribute writer method ignores a non-nil value' do
+    context 'when the attribute has a reader method' do
+      it 'raises a CouldNotSetAttribute error' do
+        model = define_active_model_class 'Example' do
+          attr_reader :name
+
+          def name=(value)
```

Comment:
Unused method argument - `value`. If it's necessary, use `_` or `_value` as an argument name to indicate that it won't be used. You can also write as `name=(*)` if you want the method to accept any arguments but don't care about them.